src/test/java/tests/SentinelTests.java

```diff
@@ -20,14 +19,21 @@
 	, plugin = {"json:target/cucumber.json",
 			"com.aventstack.extentreports.cucumber.adapter.ExtentCucumberAdapter:"}
 	, strict = true
-//  , tags = { "@text" }
+  //, tags = { "@47" }
 )
 
 public class SentinelTests {
     private static final Logger log = LogManager.getLogger(SentinelTests.class); // Create a logger.
 
+    @BeforeClass
+    public static void setUpBeforeClass() {
+        WebDriverFactory.instantiateWebDriver();
+        if(Configuration.toString("addVideoToReport")!=null)
```

Comment:
Put this in a @Before step that will run before each Scenario, and then see if you can read a @record tag for the Cucumber step to turn this on. Then this would be on if either the `addVideoToReport` option is on OR a scenario is tagged AND the record is not explicitly set to off.

If @record tag exists and `addVideoToReport` is not set (null) = record
If @record tag doesn't exist and `addVideoToReport` is set true = record
If @record tag exists and `addVideoToReport` is set true = record
If @record tag exists and `addVideoToReport` is set off/false = DON'T record
Else = DON'T record